### PR TITLE
Minor: add publishing instructions

### PR DESCRIPTION
**Pull request by alamb**
As part of https://github.com/apache/datafusion-site/pull/6 I had to figure this out, so I wanted to write it down

Eventually it would be nice to automate this process but for now at least we can have it documented

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -38,4 +38,30 @@ bundle exec jekyll serve
 
 ## Publish site
 
-TBD, but create the html with `bundle exec jekyll build` then check the content into `asf-site` branch
\ No newline at end of file
+This is currently a manual process. Basic steps are:
+
+#### Check out `main` and build site
```
Comment (phillipleblanc) [inline README.md:43]: We have had success using the https://github.com/peaceiris/actions-gh-pages action to automate publishing static content into another branch for our docs. It's built for GitHub Pages, but setting `publish_branch` to `asf_site` and `publish_dir` to `_site` might be enough to make it work for this. Here's [our workflow] if you want to take a look.

[our workflow]: https://github.com/spiceai/docs/blob/trunk/.github/workflows/build_and_publish.yml

**Review by phillipleblanc — APPROVED** [review]

**Review by andygrove — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (alamb) [reply README.md:43]: Thank you @phillipleblanc  -- the arrow blog has an automated workflow too which I think we could take a friendly look at as well.

https://github.com/apache/arrow-site/blob/main/.github/workflows/deploy.yml

I will file a ticket to track the idea

File: README.md [hunk 2]
```diff
@@ -38,4 +38,30 @@ bundle exec jekyll serve
...
+Checkout a separate copy of `datafusion-site`
+
+```shell
+git checkout asf-site
+git pull
+# create a branch for the publishing
+git checkout -b publish_blog
+# copy content built from _site directory
+cp -R ../datafusion-site/_site/* .
+git commit -a -m 'Publish blog content'
```
Comment (comphead) [inline README.md:62]: if we do a commit should it be followed by push?

**Review by comphead — APPROVED** [review]
lgtm thanks @alamb

File: README.md (hunk 1 above)
Comment (alamb) [reply README.md:43]: Filed https://github.com/apache/datafusion-site/issues/13

File: README.md (hunk 2 above)
Comment (alamb) [reply README.md:62]: Yes, good call -- will fix

Comment (alamb) [reply README.md:62]: in https://github.com/apache/datafusion-site/pull/14

**Pull request merged**